### DISCUSSION TITLE
Email verification with linked OAuth

### DISCUSSION THREAD

**Discussion by thejoyfuldev** [pocketbase/pocketbase#1353] [Q&A] [answered]
My app uses pb's magic link to ensure the user is a valid company employee.  However, I want them afterwards to be able to link an OAuth account so they aren't required to have access to the email all them time for a magic link.  Is there a way to link these two accounts together?

**Comment by ganigeorgiev** [comment, marked as answer]
The OAuth2 account will be automatically linked with the existing user if:
- is not already linked to another PocketBase user
- the provider returns an email that match with the PocketBase user

Additionally, if the provider doesn't return an email, you can also "enforce" linking by first authenticating the account as usual (eg. email and pass) and then send the OAuth2 auth request, aka.:

```js
await pb.collection('users').authWithPassword('test@example.com', '123456');

// send OAuth2 auth request with valid Authorization header set from the previous request
await pb.collection('users').authWithOAuth2(...)
```
Note that the above will work only if:
- the OAuth2 user is not already linked to another PocketBase user
- both auth requests are from the same auth collection (in the above example - `users`).

**Reply by thejoyfuldev** [reply]
So, no method of manually linking accounts where email addresses do not match?  Plan is that the user would only be able to link after being authenticated so the users record would already be available to the auth flow.

**Reply by ganigeorgiev** [reply]
> So, no method of manually linking accounts where email addresses do not match?

The second provided approach (with the 2 auth requests) is used for that.

If that's not what you are looking for, could you elaborate a little more on your use case and expected flow?

**Reply by thejoyfuldev** [reply]
Wow, completely misread that line. Yes, that will work.
